datasets/multilingual_librispeech/multilingual_librispeech.py

```diff
@@ -0,0 +1,153 @@
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "file": datasets.Value("string"),
+                    "audio": datasets.features.Audio(sampling_rate=16_000),
+                    "text": datasets.Value("string"),
+                    "speaker_id": datasets.Value("int64"),
+                    "chapter_id": datasets.Value("int64"),
+                    "id": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=("file", "text"),
+            homepage=_URL,
+            citation=_CITATION,
+            task_templates=[AutomaticSpeechRecognition(audio_file_path_column="file", transcription_column="text")],
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(self.config.data_dir)
+        data_path = os.path.join(archive_path, "mls_" + self.config.name)
+
+        train_splits = [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN, gen_kwargs={"data_dir": os.path.join(data_path, "train")}
+            ),
+            datasets.SplitGenerator(
+                name="train.9h",
+                gen_kwargs={"data_dir": os.path.join(data_path, "train"), "sub_folder": "limited_supervision/9hr"},
+            ),
+            datasets.SplitGenerator(
+                name="train.1h",
+                gen_kwargs={"data_dir": os.path.join(data_path, "train"), "sub_folder": "limited_supervision/1hr"},
+            ),
+        ]
+
+        return train_splits + [
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION, gen_kwargs={"data_dir": os.path.join(data_path, "dev")}
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST, gen_kwargs={"data_dir": os.path.join(data_path, "test")}
+            ),
+        ]
+
+    def _generate_examples(self, data_dir, sub_folder=""):
+        """Generate examples from a Multilingual LibriSpeech data dir."""
+        transcript_path = os.path.join(data_dir, "transcripts.txt")
+        key = 0
+
+        all_ids = None
+        if sub_folder != "":
+            sub_path = os.path.join(data_dir, sub_folder)
+            all_ids_paths = glob.glob(sub_path + "/*/*.txt") + glob.glob(sub_path + "/*.txt")
+            all_ids = []
+            for path in all_ids_paths:
+                with open(path, "r", encoding="utf-8") as f:
+                    all_ids += [l.strip() for l in f.readlines()]
+
+            all_ids = set(all_ids)
+
+        with open(transcript_path, "r", encoding="utf-8") as f:
+            for line in f:
+                line = line.strip()
+                id_, transcript = line.split("\t")
+
+                if all_ids is not None and id_ not in all_ids:
+                    # this only holds true for train.9h and train.1h
+                    continue
```

Comment:
Therefore "train" split also contains ""train.9h" and "train.1h"?

Reply:
yes

Reply:
just like train.100 and train.360 for datasets